src/type/matrix/SparseMatrix.js

```diff
@@ -721,12 +721,26 @@ export const createSparseMatrixClass = /* #__PURE__ */ factory(name, dependencie
 
     // check sizes
     size.forEach(function (value) {
-      if (!isNumber(value) || !isInteger(value) || value < 0) {
-        throw new TypeError('Invalid size, must contain positive integers ' +
-                            '(size: ' + format(size) + ')')
+      if (!isNumber(value) || !isInteger(value) || value < -1) {
+        throw new TypeError('Invalid size, must contain positive integers or -1 ' +
+          '(size: ' + format(size) + ')')
       }
     })
 
+    const wildCardIndex = size.indexOf(-1)
+    if (size.indexOf(-1, wildCardIndex + 1) >= 0) {
+      throw new Error('More than one wildcard in size')
+    }
+    if (wildCardIndex >= 0) {
+      const newLength = size[0] * size[1]
+      const currentLength = this._size[0] * this._size[1]
+      if (currentLength % newLength !== 0) {
+        throw new Error('Could not replace wildcard, since ' + currentLength + ' is no multiple of ' + (-newLength))
+      } else {
+        size[wildCardIndex] = -currentLength / newLength
```

Comment:
I think it would be good not to touch `size` itself, but instead create a copy of the size into say `resolvedSize` and make changes there instead.

Reply:
I thought about it. 
But there is one line that prevented me from doing that easily: 
https://github.com/josdejong/mathjs/blob/dcab05272a6d5582de8c68985399779c82b47222/src/type/matrix/DenseMatrix.js#L461

The DenseMatrix sets the size as its dimensions internally. That's why I opted to replace the wildcard with a value (since that `-1` has no other use than to be replaced).